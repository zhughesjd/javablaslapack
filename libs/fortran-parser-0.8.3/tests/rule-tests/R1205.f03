! Test interface-body
!      interface-body  is  function-stmt
!                            [ specification-part ]
!                          end-function-stmt
!                      or  subroutine-stmt
!                            [ specification-part ]
!                          end-subroutine-stmt
!
! Not tested here: function-stmt, specification-part, end-function-stmt, 
! subroutine-stmt, and end-subroutine-stmt.
!
! interface-body tested as part of an interface-block.

! Test functions.
interface
   function foo()
   end function foo
end interface

! Test subroutines.
interface
   subroutine sub()
   end subroutine sub
end interface

! Test both together; this is not a test of an interface-body, but actually
! a test of interface-specification, which allows for multiple interface-body.
interface
   subroutine sub()
   end subroutine sub
   function foo()
   end function foo
end interface

end

   
