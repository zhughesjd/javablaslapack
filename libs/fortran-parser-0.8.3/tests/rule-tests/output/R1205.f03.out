*******************************************
args[1]: R1205.f03
R1101:main-program__begin:
R1203:interface-stmt__begin:
R1203:interface-stmt: abstract_token=null hasGenericSpec=false
R1224:function-stmt__begin:
R1224:function-stmt: name=[@34,520:522='foo',<209>,16:12] hasGenericNameList=false hasSuffix=false
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R1230:end-function-stmt: name=[@45,542:544='foo',<209>,17:16]
R1205:interface-body: hasPrefix=true
R1202:interface-specification:
R1204:end-interface-stmt: hasGenericSpec=false
R1201:interface-block:
R 207:declaration-construct:
R1203:interface-stmt__begin:
R1203:interface-stmt: abstract_token=null hasGenericSpec=false
R1232:subroutine-stmt__begin:
R1232:subroutine-stmt: name=[@61,605:607='sub',<209>,22:14] hasPrefix=false hasDummyArgList=false hasBindingSpec=false hasArgSpecifier=true
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R1234:end-subroutine-stmt: name=[@72,629:631='sub',<209>,23:18]
R1205:interface-body: hasPrefix=false
R1202:interface-specification:
R1204:end-interface-stmt: hasGenericSpec=false
R1201:interface-block:
R 207:declaration-construct:
R1203:interface-stmt__begin:
R1203:interface-stmt: abstract_token=null hasGenericSpec=false
R1232:subroutine-stmt__begin:
R1232:subroutine-stmt: name=[@90,827:829='sub',<209>,29:14] hasPrefix=false hasDummyArgList=false hasBindingSpec=false hasArgSpecifier=true
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R1234:end-subroutine-stmt: name=[@101,851:853='sub',<209>,30:18]
R1205:interface-body: hasPrefix=false
R1202:interface-specification:
R1224:function-stmt__begin:
R1224:function-stmt: name=[@108,867:869='foo',<209>,31:12] hasGenericNameList=false hasSuffix=false
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R1230:end-function-stmt: name=[@119,889:891='foo',<209>,32:16]
R1205:interface-body: hasPrefix=true
R1202:interface-specification:
R1204:end-interface-stmt: hasGenericSpec=false
R1201:interface-block:
R 207:declaration-construct:
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=3
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=false hasInternalSubprogramPart=false
Parser exiting normally
********************************************
